src/test/java/api/items/ItemApiExamples.java

```diff
@@ -1144,6 +1152,84 @@ public void canUpdateAnItemWithExistingCirculationNote()
     selfLinkShouldBeReachable(updatedItem);
   }
 
+  @Test
+  public void canPopulateLocationProperties() throws Exception {
+    UUID itemId = UUID.randomUUID();
+
+    JsonObject newItemRequest = new JsonObject()
+      .put("id", itemId.toString())
+      .put("holdingsRecordId", createInstanceAndHolding().toString())
+      .put("materialTypeId", getDvdMaterialType())
+      .put("permanentLoanTypeId", getCanCirculateLoanType())
+      .put("permanentLocationId", getMainLibraryLocation())
+      .put("temporaryLocationId", getReadingRoomLocation())
+      .put("effectiveLocationId", getReadingRoomLocation());
```

Comment:
Hi @marcjohnson-kint, since we are using fake storage modules there is no ability to emulate triggers behavior. And this test verifies that effectiveLocationId is transferred from storage response and converted to `effectiveLocation` property (it is a object with `id` and `name` properties)

Reply:
@bohdan-suprun 

From my perspective, this is one of the challenges with moving business logic to the storage modules. It increases the coupling between the business logic and storage modules, and creates a reliance on that behaviour.

My concern is that this approach is not representative of what will actually happen. I think that impedes it's ability to verify the expected behaviour.

To me, what this test demonstrates is that the business logic provides a property to storage and gets it's back, which I think is quite different to relying on the storage determining that property.

I agree that this increases the complexity needed within the fakes (in many ways, they represent the degree of coupling between the modules). 

Personally, I think that complexity needs to be represented in the tests, which might mean expanding the capabilities of the fakes or using a custom one.